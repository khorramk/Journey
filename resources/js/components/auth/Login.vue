<template>
    <div class="login-wrapper">
        <form  v-on:submit.prevent="login">
            <label for="email" class="col-md-4 col-form-label text-md-right">email address</label>
            <input id="email" type="email" class="form-control " v-model="email" name="email" value="old email" required autocomplete="email" autofocus>
            <div v-if="error"  class="wrapper">
                <span class="invalid-feedback" role="alert">
                <strong>{{ this.$data.fallback }}</strong>
                </span>
            </div>
            <label for="password" class="col-md-4 col-form-label text-md-right">Password</label>
                <input id="password" v-model="password" type="password" class="form-control " name="password" required autocomplete="current-password">
                <div  v-if="error" class="wrapper">
                    <span class="invalid-feedback" role="alert">
                    <strong>{{ this.$data.fallback }}</strong>
                    </span>
                </div>

                <input class="form-check-input" type="checkbox" name="remember" id="remember">
                <label class="form-check-label" for="remember">
                <h2>remember me</h2>
            </label>
            <button type="submit" class="btn btn-primary">
                login
            </button>
            <!-- @if (Route::has('password.request'))
            <a class="btn btn-link" href="{{ route('password.request') }}">
            {{ __('Forgot Your Password?') }}
            </a>
            @endif -->
        </form>
        <a href="/register">Register</a>
    </div>
   

</template>

<script>
    export default {
        data() {
            return {
                error: false,
                email: '',
                password: '',
                fallback: ''
            }
        },
        methods: {
            login() {
                axios.post('/login', {
                    email: this.$data.email,
                    password: this.$data.password,

                })
                    .then(()=> window.location.href = '/dashboard').catch(()=> {
                        this.$data.error = true;
                        this.$data.fallback = 'invalid auth';
                    
                    })
            }
        },
    }
</script>

<style lang="scss" scoped>

</style>