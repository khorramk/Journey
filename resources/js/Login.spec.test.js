import { mount } from '@vue/test-utils';
import Vue from 'vue';
import axios from 'axios';
import VueRouter from 'vue-router';
import Login from './components/auth/Login';
import expectExport from 'expect';


const wrapper = mount(Login);

const vm = wrapper.vm;

console.log(wrapper);

it('should render', ()=> {
    expect(wrapper.element).toMatchSnapshot();
})
