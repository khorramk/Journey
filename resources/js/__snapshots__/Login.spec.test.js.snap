// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<form
  class="login-wrapper__form-login-contianer login-form bg-pale-green"
>
  <div
    class="login-input-container flex flex-col "
  >
    <label
      class="login-email_label label-spacing"
      for="email"
    >
      Email
    </label>
     
    <input
      autocomplete="email"
      autofocus="autofocus"
      class="login-inputs-container__login-email-input login-input shadow-md  border-4  border border-4  font-bold border-black"
      id="email"
      name="email"
      placeholder="Email"
      required="required"
      type="email"
      value="old email"
    />
     
    <!---->
     
    <label
      class="login-inputs-container label-spacing"
      for="password"
    >
      Password
    </label>
     
    <input
      autocomplete="current-password"
      class="login-input-passsword  shadow-md border-4  border border-4 login-input shadow-inside-shadow font-bold border-black"
      id="password"
      name="password"
      placeholder="Password"
      required="required"
      type="password"
    />
     
    <!---->
  </div>
   
  <div
    class="btn-submits-container relative"
  >
    <button
      class="submit-redirection__btn-login btn-submit-login  "
      type="submit"
    >
        
        login
        
    </button>
     
    <a
      class="submit-redirection__link btn-link-register"
      href="/register"
    >
      Register
    </a>
  </div>
</form>
`;
